### PR TITLE
Installation script changes to incorporate private ca

### PR DESCRIPTION
**Pull request by shankgan**

### CODE REVIEW

File: scripts/asm-installer/install_asm [hunk 1]
```diff
@@ -450,7 +456,11 @@ OPTIONS:
                                       omitted if not installing optional
                                       features. To add multiple files, specify
                                       them with multiple options one at a time.
-
+  --ca_id|--ca-id         <CA_ID>     Required only if --ca option is private_ca.
```
Comment (pmansour) [inline scripts/asm-installer/install_asm:459]: a note on terminology - the resource *ID* is just the last part of the full string, while the resource *name* is the `projects/foo/locations/bar/certificateAuthorities/baz` string you described below. In this case, I think they are flipped around.

more context: https://google.aip.dev/122

File: scripts/asm-installer/install_asm [hunk 2]
```diff
@@ -1598,6 +1627,13 @@ install_secrets() {
     --from-file="${CA_CHAIN}"
 }
 
+init_private_ca() {
+  WORKLOAD_IDENTITY="$PROJECT_ID.svc.id.goog[istio-system/istiod-service-account]"
+  retry 3 run gcloud projects add-iam-policy-binding $PROJECT_ID \
```
Comment (pmansour) [inline scripts/asm-installer/install_asm:1632]: (discussed offline yesterday) we can now add this binding on the CA itself

File: scripts/asm-installer/install_asm (hunk 1 above)
Comment (shankgan) [reply scripts/asm-installer/install_asm:459]: Thanks for the clarification! Will rename it to resource name

File: scripts/asm-installer/install_asm (hunk 2 above)
Comment (myidpt) [reply scripts/asm-installer/install_asm:1632]: @pmansour what does that mean? We still using the IAM binding, and the GCP SA to call the CAS, right?

Comment (pmansour) [reply scripts/asm-installer/install_asm]: We plan to use the 3p identity directly to call the CAS, but what I meant here is that we can grant the IAM role on the CA resource directly (vs. granting it on the parent project resource). This just means a more granular IAM binding which is always good for security :)

Comment (myidpt) [reply scripts/asm-installer/install_asm]: Thanks for the clarification.

**Review by myidpt — APPROVED** [review]

**Comment by shankgan** [comment]
@zerobfd - Could you review the changes?

**Review by zerobfd — APPROVED** [review]

**Pull request merged**